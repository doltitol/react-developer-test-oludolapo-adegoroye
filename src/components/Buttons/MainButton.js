import React, { PureComponent } from 'react';
import './buttons.style.scss';

export class MainButton extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return (
      <button
        className='main-button'
        style={{
          width: this.props.width,
          border: this.props.outline
            ? '1px solid #1D1F22'
            : this.props.disabled
            ? 'rgba(94, 206, 123, 0.6)'
            : '1px solid #5ECE7B',
          height: this.props.height,
          color: this.props.outline ? '#1D1F22' : '#FFFFFF',
          background: this.props.outline
            ? 'transparent'
            : this.props.disabled
            ? 'rgba(94, 206, 123, 0.6)'
            : '#5ECE7B',
        }}
        onClick={this.props.onClick}
        disabled={this.props.disabled}
      >
        {this.props.text}
      </button>
    );
  }
}

export default MainButton;
