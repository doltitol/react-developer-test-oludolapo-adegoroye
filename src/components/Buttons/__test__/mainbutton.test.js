import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MainButton from '../MainButton';

describe('<MainButton /> spec', () => {
  it('renders the component', () => {
    const container = render(<MainButton text='hello' />);
    const getByTestId = container.getByTestId;
    const mainButton = getByTestId('main-button');
    expect(mainButton.textContent).toEqual('hello');
    expect(mainButton).toHaveStyle('background: rgb(94, 206, 123)');
    expect(container.children).toMatchSnapshot();
  });
});
