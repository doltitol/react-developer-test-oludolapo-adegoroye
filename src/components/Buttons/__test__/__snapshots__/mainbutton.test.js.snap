// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainButton /> spec renders the component 1`] = `undefined`;

exports[`<MainButton /> spec renders the empty component 1`] = `undefined`;
